test/test_torch.py

```diff
@@ -2175,41 +2175,72 @@ def test_lu(self):
 
     @staticmethod
     def _test_lu_solve(self, cast, pivot=True):
-        a = torch.FloatTensor((((1.3722, -0.9020),
-                                (1.8849, 1.9169)),
-                               ((0.7187, -1.1695),
-                                (-0.0139, 1.3572)),
-                               ((-1.6181, 0.7148),
-                                (1.3728, 0.1319))))
-        b = torch.FloatTensor(((4.02, 6.19),
-                               (-1.56, 4.00),
-                               (9.81, -4.09)))
-        a, b = cast(a), cast(b)
-        LU_data, pivots, info = a.lu(get_infos=True, pivot=pivot)
-        self.assertEqual(info.abs().sum(), 0)
-        with warnings.catch_warnings(record=True) as w:
-            x = torch.lu_solve(b, LU_data, pivots)
-            b_ = torch.bmm(a, x.unsqueeze(2)).squeeze()
-            self.assertEqual(b_, b)
-            # Warning related check
-            self.assertEqual(len(w), 1)
-            self.assertTrue("Passing RHS tensor with number of dimensions = 2 is deprecated" in str(w[-1].message))
-
         from common_utils import random_fullrank_matrix_distinct_singular_value
-        for batches in [2, 5]:
-            for k, n in zip([2, 3, 5], [3, 5, 7]):
-                A = cast(random_fullrank_matrix_distinct_singular_value(n, batches))
-                b = cast(torch.randn(batches, n, k))
-                LU_data, LU_pivots, infos = A.lu(get_infos=True)
-                self.assertEqual(infos, torch.zeros_like(infos))
-                x = torch.lu_solve(b, LU_data, LU_pivots)
-                b_ = torch.matmul(A, x)
-                self.assertEqual(b_, b)
+
+        for k, n in zip([2, 3, 5], [3, 5, 7]):
+            A = cast(random_fullrank_matrix_distinct_singular_value(n))
+            b = cast(torch.randn(n, k))
+            LU_data, LU_pivots, info = A.lu(get_infos=True, pivot=pivot)
+            self.assertEqual(info, torch.zeros_like(info))
+            x = torch.lu_solve(b, LU_data, LU_pivots)
+            b_ = torch.matmul(A, x)
+            self.assertEqual(b_, b)
 
     @skipIfNoLapack
     def test_lu_solve(self):
         self._test_lu_solve(self, lambda t: t)
 
+    @staticmethod
+    def _test_lu_solve_batched(self, cast, pivot=True):
```

Comment:
Shouldn't pivot always be True? This test doesn't really make sense with a `pivot` parameter

Reply:
When `pivot` is `False`, an identify permutation is returned. `pivot` doesn't directly play a role in `lu_solve`, but plays a role in the `lu` decomposition required for `lu_solve`.

Reply:
Oh, I see. Is it intentional that the test is never run with `pivot=False`?

Reply:
The test with `pivot=False` is enabled by CUDA alone, since only CUDA has the non-pivoting variant. This is there in `test_cuda.py`

Reply:
Ah, sorry that I missed that. Thank you for the clarification!